action-ssh-to-runner/action.yaml

```
@@ -17,27 +17,70 @@ inputs:
 runs:
   using: "composite"
   steps:
-    - name: "SSH to runner: Package dependencies for tmate"
+    - name: "SSH to runner: Write tmate config file"
       shell: bash
       run: |
-        ${{ inputs.sudo }} && sudo_cmd="sudo" || sudo_cmd=""
-        # mxschmitt/action-tmate does not have a check to use yum/dnf instead of apt.
-        # For now, all dependencies needed by tmate are already installed.
-        test -f /etc/lsb-release || $sudo_cmd ln -s /bin/true /bin/apt-get
-    - name: "SSH to runner: Print usage information"
+        TMATE_CONF=/tmp/tmate.conf
+        echo "TMATE_CONF=$TMATE_CONF" >> $GITHUB_ENV
+        echo "set -g tmate-server-host ${{ inputs.tmate-server-host }}" >> ${TMATE_CONF}
+        echo "set -g tmate-server-port ${{ inputs.tmate-server-port }}" >> ${TMATE_CONF}
+        echo "set -g tmate-server-rsa-fingerprint ${{ inputs.tmate-server-rsa-fingerprint }}" >> ${TMATE_CONF}
+        echo "set -g tmate-server-ed25519-fingerprint ${{ inputs.tmate-server-ed25519-fingerprint }}" >> ${TMATE_CONF}
+        cat ${TMATE_CONF}
+    - name: "SSH to runner: Install tmate"
       shell: bash
       run: |
-        echo "You can connect to this runner using the ssh command printed in the next step."
-        echo "To connect, you need the ssh key used by your GitHub account."
-        echo "You also need to connected to the VPN."
-        echo "The workflow has been paused."
-        echo "To resume, create a file named 'continue' under $GITHUB_WORKSPACE or under / in the runner."
+        TMATE_VERSION=2.4.0
+        ARCH=i386
+        wget https://github.com/tmate-io/tmate/releases/download/${TMATE_VERSION}/tmate-${TMATE_VERSION}-static-linux-${ARCH}.tar.xz -P /tmp
+        tar -xf /tmp/tmate-${TMATE_VERSION}-static-linux-${ARCH}.tar.xz -C /tmp
+        TMATE_BIN=/tmp/tmate-${TMATE_VERSION}-static-linux-${ARCH}/tmate
+        echo "TMATE_BIN=$TMATE_BIN" >> $GITHUB_ENV
+    - name: "SSH to runner: Set up scripts"
+      shell: bash
+      run: |
+        # Adding a system path: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-system-path
+        # does not work now, maybe our runner version is old and does not support this functionality.
+        # echo "${GITHUB_ACTION_PATH}" >> $GITHUB_PATH
+        # Workaround, copy script to /home/runner/.local/bin
+        mkdir -p /home/runner/.local/bin
+        cp ${GITHUB_ACTION_PATH}/disconnect /home/runner/.local/bin/
+        cp ${GITHUB_ACTION_PATH}/continue_workflow /home/runner/.local/bin/
```

Unfortunately not all home are the same (mistake on our end with self-hosted)
```suggestion
        mkdir -p ${HOME}/.local/bin
        cp ${GITHUB_ACTION_PATH}/disconnect ${HOME}/.local/bin/
        cp ${GITHUB_ACTION_PATH}/continue_workflow ${HOME}/.local/bin/
```